### ISSUE TITLE
Inline chunk dependency imports to avoid chunk latency waterfall

### ISSUE THREAD

**Issue by guybedford** [rollup/rollup#2071]
The other performance issue we have with code splitting is the latency waterfall caused by loading `chunkA` which depends on `chunkB` which depends on `chunkC`, which depends on `external` as the dependencies are only discovered as loading happens.

What we should do here is ensure that the imports of `chunkA` inline the full imports down the chunking graph to `external` in order to ensure that we can load all dependencies in a maximum of two round trips always.

In short, we should inline the execution order for each entry point into the entry point itself.

So basically `entryA` wouldn't just contain `import {thing} from './chunkA.js'`, it would contain `import 'external'; import './chunkC'; import './chunkB'; import {thing} from './chunkA.js'`.

Again this shouldn't be too tricky, just tracking it here.

**Comment by lukastaegert** [comment]
You could consider using Githubs auto-closing feature by adding lines like "closes #2701" to your PR descriptions or to individual commit descriptions to avoid having to close issues manually: https://help.github.com/articles/closing-issues-using-keywords

That way, closing will be synced with merging.
